lib/org/chromium/webidl/Diff.js

```diff
@@ -131,32 +259,32 @@ foam.CLASS({
         }
 
         // Determine types and whether a direct comparison can be done.
-        var leftType = typeof(left);
-        var rightType = typeof(right);
-        var isLeftArray = Array.isArray(left);
-        var isRightArray = Array.isArray(right);
-        foam.assert(leftType === rightType && isLeftArray === isRightArray,
+        var leftType = foam.typeOf(left);
+        var rightType = foam.typeOf(right);
+        foam.assert(leftType === rightType,
             'Diff: Encountered a type mismatch!');
 
         // Type is a basic type, perform simple comparison.
-        if (['boolean', 'number', 'string'].includes(leftType)) {
+        if ([foam.Boolean, foam.Number, foam.String].includes(leftType)) {
```

Comment:
This will create garbage on every function call. Can we just use ` ... === ... ||  ... === ...`?